src/main/java/core/basesyntax/serviceimpl/ReaderServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.service.ReaderService;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReaderServiceImpl implements ReaderService<List<String>> {
+    private final List<String> lines = new ArrayList<>();
```

Comment:
The 'lines' list should not be a field of the ReaderServiceImpl class. This can lead to unexpected behavior when the 'readFromFile' method is called multiple times. Consider moving the 'lines' list inside the 'readFromFile' method to ensure that a fresh list is used for each read operation.